src/shared/tpm2-util.c

```diff
@@ -476,6 +476,16 @@ void tpm2_pcr_mask_to_selection(uint32_t mask, uint16_t bank, TPML_PCR_SELECTION
         };
 }
 
+#define tpm2_log_debug_digest(digest, fmt, ...)                         \
+        ({                                                              \
+                if (DEBUG_LOGGING) {                                    \
+                        _cleanup_free_ char *_h_ = hexmem((digest)->buffer, (digest)->size); \
+                                                                        \
+                        if (_h_)                                        \
+                                log_debug(fmt ": %s", ##__VA_ARGS__, _h_); \
+                }                                                       \
+        })
```

Comment:
i am surprised this works. i had assumed that `({ })` would have to resolve to a single type always, so that they are usually as part of expressions..

Reply:
i'll update it in the next push, thanks